src/Wallet/Wallet.ts

```diff
@@ -193,25 +194,41 @@ export default class Wallet {
     });
   }
 
+  sendingTxInProgress = false;
+
   /**
    * sendFromIndex will send a transaction from the account index specified
    * @param {number} idx The index of the account to send a transaction from.
    * @param {TransactionParams} opts {to, value, gas, gasPrice, data}
    * @returns {Promise<string>} A promise which will resolve to the transaction hash
    */
   async sendFromIndex(idx: number, opts: any) {
-    console.log('sendFromIndex', idx, opts);
     if (idx >= this.length) {
       throw new Error('Index is outside range of addresses.');
     }
     const from = this.getAccounts()[idx].getAddressString();
     const nonce = await this.getNonce(from);
-    
+
     const signedTx = await this.signTransaction(from, nonce, opts);
-    const hash = await this.sendRawTransaction(signedTx);
-    
-    const receipt = await this.getTransactionReceipt(hash, from);
-    
+
+    if (this.sendingTxInProgress) {
+      throw 'Sending transaction is already in progress. Please wait.';
+    }
+
+    let receipt;
+    try {
+      this.sendingTxInProgress = true;
```

Comment:
Instead of a single boolean we need to hold on Object of addresses => booleans. The idea of the wallet is that we can have multiple addresses contained within one Wallet object, each sending out transaction in parallel. Under this implementation, the wallet would freeze until the receipt was obtained, not allowing the other accounts to send transactions.